components/layout/layout_impl.rs

```diff
@@ -212,7 +212,8 @@ impl Drop for LayoutThread {
             .font_context
             .collect_unused_webrender_resources(true /* all */);
         self.compositor_api
-            .remove_unused_font_resources(keys, instance_keys)
+            .remove_unused_font_resources(keys, instance_keys);
+        StyleThreadPool::shutdown();
```

Comment:
I'm worried that this StyleThreadPool global for all script threads in a process. I think we need a separate place to call this from.

Reply:
Yeah, I've tested it and it is shut down multiple times, once per tab, but layout still seems to work. I guess it would be better to do this one per process shutdown.

Reply:
Ok the latest shuts down the style pool either when the constellation exits, in single process mode, or when a content process exits. I've tested it with opening and closing tabs in both modes. Thanks!